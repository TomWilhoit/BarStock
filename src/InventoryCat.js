import React, { Component } from "react";
import InventoryProd from "./InventoryProd.js";
import "./css/InventoryCat.scss";

class InventoryCat extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  displayProds = () => {
    this.props.toggleCat(this.props.category);
  };

  render() {
    if (this.props.category === this.props.selectedCat) {
      return (
        <div className="liquorProd open" onClick={this.displayProds}>
          <li className="Category">
            <h2 className="Category-titles">{this.props.category}</h2>
          </li>
          <InventoryProd
            allProducts={this.props.products}
            category={this.props.category}
            changeCart={this.props.changeCart}
            cartItems={this.props.cartItems}
          />
        </div>
      );
    } else {
      return (
        <div className="liquorProd closed" onClick={this.displayProds}>
          <li className="Category">
            <h2 className="Category-titles">{this.props.category}</h2>
          </li>
        </div>
      );
    }
  }
}

export default InventoryCat;
