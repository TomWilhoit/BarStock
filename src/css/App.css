/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

