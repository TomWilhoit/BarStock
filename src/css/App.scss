/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
