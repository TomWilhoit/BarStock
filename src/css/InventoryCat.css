
.Category-display ul {
    margin: auto;
    width: 80%;
    padding: 1em 0;
}

.Category {
    list-style-type: none;
    text-align: center;
    padding: .5em;
    /* color: #fff; */
    color: #EEAD0E;
    font-family: 'Roboto Condensed', sans-serif;
    /* background: #231f20; */
    background: #ffffff;
    margin: 0;
    text-transform: uppercase;
    font-weight: bold;
    font-size: 1.1em;
    letter-spacing: .07em;
}

.liquorProd {
    overflow: hidden;
    margin-top: .5em;
    box-shadow: 0px 2px 7px 2px rgba(35, 31, 32, .2)
}

.liquorProd:first-child {
    margin:0;
}

.liquorProd span { 
    text-align: left;    
}

.liquorProd.closed {
    max-height: 37px;
    height: auto;
    transition: height 2s;
}

.liquorProd.open {
    background: #fff;
    max-height: 500px;
    height: auto;
    transition: max-height .5s linear;
    
    /* transition: max-height 0.5s ease-out; */
}

