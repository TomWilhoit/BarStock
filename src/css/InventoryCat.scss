/* #region Liquor Product Container */
.liquorProd {
  overflow: hidden;
  margin-top: .5em;
  box-shadow: 0px 2px 7px 2px rgba(35, 31, 32, .2);
}

.liquorProd:first-child {
    margin:0;
}

.liquorProd.closed {
    max-height: 49px;
    height: auto;
    transition: height 2s;
    border-radius: 5px
}

.liquorProd.open {
    background: #fff;
    max-height: 300px;
    height: auto;
    transition: max-height .5s linear;
    border-top-right-radius: 5px;
    border-top-left-radius: 5px;
}
/* #endregion Liquor Product Container */


/* #region Cateogry Title */
.Category-display ul {
    margin: auto;
    padding: 2em;
}

.Category {
  list-style-type: none;
  padding: .7em;
  background: #11212a;
  margin: 0;
  letter-spacing: .07em;
  box-shadow: -1px 1px 2px 0px rgba(0, 0, 0, .5);
}

.Category .Category-titles {
  text-align: center;
  margin: 0;
  text-transform: uppercase;
  font-weight: bold;
  line-height: 1;
  color: #fff;
  font-size: 1.3em;
  font-family: 'Abel', sans-serif;
}
/* #endregion Cateogry Title */


/* #region products header titles */
.product-header {
  font-weight: 100;
  font-size: .7em;
  display: flex;
  justify-content: space-evenly;
  background: #fff9e8;
  box-shadow: 0px 0px 1px 0px rgba(35, 31, 32, .4);
}

.product-header .header-title {
  color: #4e4e4e;
  flex-grow: 1;
  padding: .5em;
  text-align: center;
  font-family: 'Lato', sans-serif;
}
/* #endregion products header titles */
