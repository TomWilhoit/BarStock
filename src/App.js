import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
