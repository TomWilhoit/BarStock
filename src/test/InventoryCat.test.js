import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import InventoryCat from '../InventoryCat';
import mockData from '../mockData'

// let this.props.category
let allInventory = mockData.distributor[0].inventory;
const mockFunc = jest.fn();



describe('InventoryCat', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <InventoryCat category={allInventory[0].category}
      products={allInventory}
      toggleCat={mockFunc}
      displayProds={mockFunc}
    />
      )
    });

    it('should match snapshot when all data is passed correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should display the products in a given category', () => {
      wrapper.find('.closed').simulate('click', {preventDefault: () => {}})
      expect(mockFunc).toBeCalled();
    })
    
  })