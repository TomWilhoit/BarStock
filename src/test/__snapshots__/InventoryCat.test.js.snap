// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryCat should match snapshot when all data is passed correctly 1`] = `
<div
  className="liquorProd closed"
  onClick={[Function]}
>
  <li
    className="Category"
  >
    <h2
      className="Category-titles"
    >
      whiskey
    </h2>
  </li>
</div>
`;
